analyzer/tests/functional/config/test_config.py

```diff
@@ -118,6 +118,102 @@ def test_only_clangsa_config(self):
+            json.dump({
+                'analyzer': [
+                    "--analyzer-config",
+                    "clangsa:track-conditions=false",
+                    "--analyzer-config",
+                    "clang-tidy:HeaderFilterRegex=.*"
+                ]}, config_f)
+
+        out, returncode = self.__run_analyze(self.config_file_json,
+                                             ["--verbose", "debug_analyzer"])
+
+        self.assertNotEqual(returncode, 1)
+        self.assertIn("track-conditions=false", out)
+        self.assertIn("{\"HeaderFilterRegex\": \".*\"}", out)
+
+    def test_config_file_and_cmd_resolution(self):
+        """
+        Test whether multiple --analyzer-config arguments from *both* a
+        CodeChecker config file and the CLI are merged, and don't overwrite one
+        another.
+        """
+        with open(self.config_file_json, 'w+',
+                  encoding="utf-8", errors="ignore") as config_f:
+            json.dump({
+                'analyzer': [
+                    "--analyzer-config",
+                    "clangsa:track-conditions=false"
+                ]}, config_f)
+
+        out, returncode = self.__run_analyze(self.config_file_json,
+                                             ["--analyzer-config",
+                                              "clang-tidy:"
+                                              "HeaderFilterRegex=.*",
+                                              "--verbose", "debug_analyzer"])
+
+        self.assertNotEqual(returncode, 1)
+        self.assertIn("track-conditions=false", out)
+        self.assertIn("{\"HeaderFilterRegex\": \".*\"}", out)
+
+    def test_cmd_multiple_analyzer_config_resolution(self):
+        """
+        Test whether multiple --analyzer-config arguments from the command line
+        are merged, and don't overwrite one another.
+        """
+
+        with open(self.config_file_json, 'w+',
+                  encoding="utf-8", errors="ignore") as config_f:
+            config_f.write("")
+
+        out, returncode = self.__run_analyze(self.config_file_json,
+                                             ["--analyzer-config",
+                                              "clang-tidy:"
+                                              "HeaderFilterRegex=.*",
+                                              "--analyzer-config",
+                                              "clangsa:track-conditions=false",
+                                              "--verbose", "debug_analyzer"])
+
+        self.assertNotEqual(returncode, 1)
+        self.assertIn("track-conditions=false", out)
+        self.assertIn("{\"HeaderFilterRegex\": \".*\"}", out)
+
+    def test_cmd_overrides_config_file(self):
+        """
+        Test the override of multiple --analyzer-config arguments from *both* a
+        CodeChecker config file and the CLI are merged, but if they specify the
+        very same config, CLI arguments override the config file.
```

Comment:
I think we need to rephrase it. A command line option doesn't override the ones In config file, but appends it to a later position in the analyzer command.